### PR TITLE
Adapting server to ParallelWaveGAN

### PR DESCRIPTION
**Pull request by erogol**

### CODE REVIEW

File: server/server.py [hunk 1]
```diff
@@ -14,30 +14,45 @@ def convert_boolean(x):
     parser.add_argument('--tts_checkpoint', type=str, help='path to TTS checkpoint file')
     parser.add_argument('--tts_config', type=str, help='path to TTS config.json file')
     parser.add_argument('--tts_speakers', type=str, help='path to JSON file containing speaker ids, if speaker ids are used in the model')
-    parser.add_argument('--wavernn_lib_path', type=str, help='path to WaveRNN project folder to be imported. If this is not passed, model uses Griffin-Lim for synthesis.')
-    parser.add_argument('--wavernn_file', type=str, help='path to WaveRNN checkpoint file.')
-    parser.add_argument('--wavernn_config', type=str, help='path to WaveRNN config file.')
+    parser.add_argument('--wavernn_lib_path', type=str, default=None, help='path to WaveRNN project folder to be imported. If this is not passed, model uses Griffin-Lim for synthesis.')
+    parser.add_argument('--wavernn_file', type=str, default=None, help='path to WaveRNN checkpoint file.')
+    parser.add_argument('--wavernn_config', type=str, default=None, help='path to WaveRNN config file.')
```
Comment (reuben) [inline server/server.py]: nit: The default is already None if you don't specify it, so there's no need to set it in these lines.

File: server/synthesizer.py [hunk 2]
```diff
@@ -45,9 +48,9 @@ def load_tts(self, tts_checkpoint, tts_config, use_cuda):
             self.input_size = len(phonemes)
         else:
             self.input_size = len(symbols)
-        # load speakers
+        # TODO: fix this for multi-speaker model - load speakers
         if self.config.tts_speakers is not None:
-            self.tts_speakers = load_speaker_mapping(os.path.join(model_path, self.config.tts_speakers))
+            self.tts_speakers = load_speaker_mapping(self.config.tts_speakers)
```
Comment (reuben) [inline server/synthesizer.py]: Thanks, nice catch.

File: server/synthesizer.py [hunk 3]
```diff
@@ -91,11 +92,26 @@ def load_wavernn(self, lib_path, model_path, model_file, model_config, use_cuda)
...
 
         check = torch.load(model_file)
-        self.wavernn.load_state_dict(check['model'])
+        self.wavernn.load_state_dict(check['model'], map_location="cpu")
         if use_cuda:
             self.wavernn.cuda()
         self.wavernn.eval()
 
+    def load_pwgan(self, lib_path, model_file, model_config, use_cuda):
+        sys.path.append(lib_path) # set this if TTS is not installed globally
```
Comment (reuben) [inline server/synthesizer.py]: This comment is talking about TTS but it's actually about the PWGAN repo right?

Comment (reuben) [reply server/synthesizer.py]: I guess it was copied from the WaveRNN one, but then that should be fixed too? :P

File: server/server.py (hunk 1 above)
Comment (erogol) [reply server/server.py]: Otherwise, it does not work if I call ```python server.py``` in my case. It says it is not defined and raises an error.

File: server/synthesizer.py (hunk 3 above)
Comment (erogol) [reply server/synthesizer.py]: yep :) copy-paste gods will know.

**Comment by reuben** [comment]
This also needs to handle actually calling the PWGAN vocoder from `tts`, loading the PWGAN AudioProcessor, etc.

**Comment by reuben** [comment]
Also, the `Synthesizer.tts` method has a lot of other differences when compared to the `tts` function here: https://colab.research.google.com/drive/1cpofjnfKSpFhiREgExENIsum4MrqxyPR#scrollTo=f-Yc42nQZG5A

So it should also be adapted.

**Comment by erogol** [comment]
> 
> 
> Also, the `Synthesizer.tts` method has a lot of other differences when compared to the `tts` function here: https://colab.research.google.com/drive/1cpofjnfKSpFhiREgExENIsum4MrqxyPR#scrollTo=f-Yc42nQZG5A
> 
> So it should also be adapted.

The reason is that the one in the notebook is more to visualize results and the one with ```Synthesizer``` is to just synthesize voice. I think they are fine this way. So you can play with the details in the notebook without going into library.

**Comment by reuben** [comment]
Yeah I guess the only thing that threw me off was the _denormalize followed by _normalize to print some information. It's just about adding a check and call to self.pwgan. If you merge this PR I can make a new one with the fix since I did it for the server package anyway.

**Comment by erogol** [comment]
It was more about to recovered the normalization differences between the TTS and vocoder models in the notebook. We can do the same in Synthesizer in the server but I assume for the server we would always train them both with the normalization same parameters. That's also why I don't import vocoder AudioProcessor in server.
